x-pack/plugins/infra/public/components/asset_details/hooks/use_tab_switcher.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState } from 'react';
+import createContainer from 'constate';
+import { useLazyRef } from '../../../hooks/use_lazy_ref';
+import type { TabIds } from '../types';
+
+export function useTabSwitcher({ intitalActiveTabId }: { intitalActiveTabId?: TabIds }) {
```

Comment:
nit: typo for `intitalActiveTabId` on the whole hook and its usage, should be `initialActiveTabId`